lib/node_modules/@stdlib/array/base/count-ifs/docs/types/index.d.ts

```diff
@@ -0,0 +1,149 @@
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns boolean indicating whether an element passes a test
+*/
+type Ternary<U> = ( value: U, index: number, arr: Collection<U> | AccessorArrayLike<U> ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns boolean indicating whether an element passes a test
+*/
+type Predicate<U> = Nullary | Unary<U> | Binary<U> | Ternary<U>;
+
+/**
+* Counts the number of elements in the input arrays which pass the tests implemented by the predicate functions.
+*
+* @param x - input array
+* @param predicate - predicate function
+* @returns result
+*
+* @example
+* function predicate( v ) {
+*     return v > 0;
+* }
+*
+* var x = [ 0, 1, 0, 1, 1 ];
+*
+* var n = countIfs( x, predicate );
+* // returns 3
+*/
+declare function countIfs<U = unknown>( x: Collection<U> | AccessorArrayLike<U>, predicate: Predicate<U> ): number;
+
+/**
+* Counts the number of elements in the input arrays which pass the tests implemented by the predicate functions.
+*
+* @param x0 - first input array
+* @param predicate0 - first predicate function
+* @param x1 - second input array
+* @param predicate1 - second predicate function
+* @returns result
+*
+* @example
+* function predicate0( v ) {
+*     return v > 0;
+* }
+*
+* function predicate1( v ) {
+*     return v < 0;
+* }
+*
+* var x0 = [ 0, 1, 0, 1, 1 ];
+* var x1 = [ -1, 2, 4, -5, -8 ];
+*
+* var n = countIfs( x0, predicate0, x1, predicate1 );
+* // returns 2
+*/
+declare function countIfs<U = unknown>( x0: Collection<U> | AccessorArrayLike<U>, predicate0: Predicate<U>, x1: Collection<U> | AccessorArrayLike<U>, predicate1: Predicate<U> ): number;
+
+/**
+* Counts the number of elements in the input arrays which pass the tests implemented by the predicate functions.
+*
+* @param x0 - first input array
+* @param predicate0 - first predicate function
+* @param x1 - second input array
+* @param predicate1 - second predicate function
+* @param x2 - third input array
+* @param predicate2 - third predicate function
+* @returns result
+*
+* @example
+* function predicate0( v ) {
+*     return v > 0;
+* }
+*
+* function predicate1( v ) {
+*     return v < 0;
+* }
+*
+* function predicate2( v ) {
+*     return v % 2 === 0;
+* }
+*
+* var x0 = [ 0, 1, 0, 1, 1 ];
+* var x1 = [ -1, -2, 4, 5, -8 ];
+* var x2 = [ 0, 4, 3, 2, 12 ];
+*
+* var n = countIfs( x0, predicate0, x1, predicate1 );
+* // returns 2
+*/
+declare function countIfs<U = unknown>( x0: Collection<U> | AccessorArrayLike<U>, predicate0: Predicate<U>, x1: Collection<U> | AccessorArrayLike<U>, predicate1: Predicate<U>, x2: Collection<U> | AccessorArrayLike<U>, predicate2: Predicate<U> ): number;
```

Comment:
You can definitely simplify this defn with `InputArray`.

Also, you stopped with predicate pairs. I suggest going up to 5 array/predicate pairs (i.e., 10 arguments). Then, for your last overload you need to use a rest parameter which is a union of `InputArray | Predicate`, so that people can provide more than 10 args and not have the compiler complain.